lib/jekyll/url.rb

```diff
@@ -141,7 +144,10 @@ def self.escape_path(path)
       #   sub-delims    = "!" / "$" / "&" / "'" / "(" / ")"
       #                 / "*" / "+" / "," / ";" / "="
       path = Addressable::URI.encode(path)
-      path.encode("utf-8").sub("#", "%23")
+      escaped_path = path.encode("utf-8").sub("#", "%23")
+      cache[raw_path] = escaped_path
+
+      escaped_path
```

Comment:
Are all the transient variables needed?

```ruby
  escaped_path = Addressable::URI.encode(path).encode("utf-8")
  cache[path] = escaped_path.sub("#", "%23")
end
```

This seems readable enough to me.

Reply:
should I not return `escaped_path` (before the closing end)?

Reply:
`cache[path] = escaped_path.sub("#", "%23")` evaluates to the right hand side of the assignment. This means:

```ruby
cache[raw_path] = escaped_path
escaped_path
```

is the same as

```ruby
cache[raw_path] = escaped_path
```

Too weird?

Reply:
a bit, yes. But I like how DRY it is..
just wasn't sure if that's the approach to take..